File: SCH/SCH.Client/src/app/sch/interfaces/teacher.ts
import { UserLookup } from './user-lookup';

export interface Teacher {
    id: number;
    name: string;
    userId?: number | null;
    user?: UserLookup | null;

    /**
     * Row version for optimistic concurrency control
     * Base64-encoded byte array from backend
     */
    rowVersion?: string;
}

File: SCH/SCH.Models/Teachers/Entities/Teacher.cs
namespace SCH.Models.Teachers.Entities
{
    using SCH.Models.Common.AuditableEntities;
    using SCH.Models.Common.ConcurrencyEntities;

    public class Teacher : IAuditableEntity, IConcurrencyEntity
    {
        public int Id { get; set; }

        public required string Name { get; set; }

        // Audit properties
        public int CreatedBy { get; set; }
        public DateTime CreatedDate { get; set; }
        public int? ModifiedBy { get; set; }
        public DateTime? ModifiedDate { get; set; }

        // Concurrency control
        /// <summary>
        /// Row version for optimistic concurrency control
        /// </summary>
        public byte[] RowVersion { get; set; } = Array.Empty<byte>();
    }
}
